File: src/main/java/com/designpatterns/dev/behavioral/state/java/Canvas.java
package com.designpatterns.dev.behavioral.state.java;

public class Canvas {
    private Tool currentTool;

    public String mouseUp(){
        return currentTool.mouseUp();
    }

    public String mouseDown(){
        return currentTool.mouseDown();
    }


    public Tool getCurrentTool() {
        return currentTool;
    }

    public void setCurrentTool(Tool currentTool) {
        this.currentTool = currentTool;
    }
}

File: src/main/java/com/designpatterns/dev/behavioral/state/kotlin/Canvas.kt
package com.designpatterns.dev.behavioral.state.kotlin

class Canvas(tool: Tool) {
       var currentTool: Tool = tool
        get() {
            return field
        }


    fun mouseUp(): String {
        return currentTool.mouseUp()
    }

    fun mouseDown(): String {
        return currentTool.mouseDown()
    }


}